src/components/Viewport.ts

```diff
@@ -1849,4 +1923,39 @@ export default class Viewport {
 
     return { removedPanels, addedPanels };
   }
+
+  private resizePanels(panels: Panel[]): void {
+    const options = this.options;
+    const panelBboxes = this.panelBboxes;
+
+    if (options.isEqualSize === true) {
+      if (!panelBboxes.default) {
+        const defaultPanel = panels[0];
+        panelBboxes.default = defaultPanel.getBbox();
+      }
+
+      const defaultBbox = panelBboxes.default;
+
+      panels.forEach(panel => {
+        panel.resize(defaultBbox);
+      });
+      return;
+    } else if (options.isEqualSize) {
+      const equalSizeClasses = options.isEqualSize as string[]; // for readability
+
+      panels.forEach(panel => {
+        const overlappedClass = panel.getOverlappedClass(equalSizeClasses);
+        if (overlappedClass) {
+          panel.resize(panelBboxes[overlappedClass]);
+          panelBboxes[overlappedClass] = panel.getBbox();
+        } else {
+          panel.resize();
+        }
+      });
+      return;
+    }
+    panels.forEach(panel => {
```

Comment:
@WoodNeck 

Why is the panel resized once more?

Reply:
There're `return`s above.